File: csharp/tests/Idkollen.Client.Tests/FtnTests.cs
using System.Net;
using Idkollen.Client.Models;
using Xunit;

namespace Idkollen.Client.Tests;

public class FtnTests
{
    private static (Idkollen.Client.IdkollenClient Client, MockHttpMessageHandler Handler) Build()
    {
        var handler = new MockHttpMessageHandler();
        var http = new HttpClient(handler);
        var client = new Idkollen.Client.IdkollenClientBuilder("cid", "sec")
            .BaseUrl("https://x.test").HttpClient(http).Build();
        return (client, handler);
    }

    [Fact]
    public async Task AuthPending()
    {
        var (client, handler) = Build();
        handler.Enqueue(HttpStatusCode.OK, new { status = "PENDING", id = "f1", url = "https://login" });
        var result = await client.Ftn.AuthAsync(new FtnAuthRequest());
        var p = Assert.IsType<FtnPending>(result);
        Assert.Equal("https://login", p.Url);
    }

    [Fact]
    public async Task AgeVerificationCompleted()
    {
        var (client, handler) = Build();
        handler.Enqueue(HttpStatusCode.OK, new { status = "COMPLETED", id = "av1", ageVerified = true });
        var result = await client.Ftn.AgeVerificationAsync(new AgeVerificationRequest { MinAge = 18 });
        var done = Assert.IsType<AgeVerificationCompleted>(result);
        Assert.True(done.AgeVerified);
    }
}

File: swift/Tests/IdkollenClientTests/FtnTests.swift
import XCTest
@testable import IdkollenClient

final class FtnTests: XCTestCase {
    var client: IdkollenClient!

    override func setUp() {
        super.setUp()
        MockURLProtocol.reset()
        client = IdkollenClientBuilder(clientId: "cid", clientSecret: "sec")
            .baseUrl("https://x.test")
            .urlSession(.mocked())
            .build()
    }

    func testAuthPending() async throws {
        MockURLProtocol.enqueue(status: 200, json: [
            "status": "PENDING",
            "id": "abc",
            "url": "https://login",
        ])

        let result = try await client.ftn.auth(FtnAuthRequest())
        guard case .pending(let p) = result else { return XCTFail("expected pending") }
        XCTAssertEqual(p.url, "https://login")
        XCTAssertEqual(p.id, "abc")
    }

    func testAgeVerificationCompleted() async throws {
        MockURLProtocol.enqueue(status: 200, json: [
            "status": "COMPLETED",
            "id": "age1",
            "ageVerified": true,
        ])

        let result = try await client.ftn.ageVerification(AgeVerificationRequest(minAge: 18))
        guard case .completed(let done) = result else { return XCTFail("expected completed") }
        XCTAssertTrue(done.ageVerified)
    }
}
